src/html/comrak_adapters.rs

```diff
@@ -232,11 +232,37 @@ pub struct ToCEntry {
   pub anchor: String,
 }
 
+#[derive(Default)]
+pub struct Anchorizer {
+  map: HashMap<String, i32>,
+  itoa_buffer: itoa::Buffer,
+}
+
+impl Anchorizer {
+  pub fn anchorize(&mut self, s: &str) -> String {
```

Comment:
Maybe let's link to comrak's anchorize document to explain what this does. Also maybe better mention we do this for perf improvement?

```suggestion
  /// Returns a String that has been converted into an anchor using the GFM algorithm.
  /// This replaces comrak's implementation to improve the performance.
  /// @see https://docs.rs/comrak/latest/comrak/struct.Anchorizer.html#method.anchorize
  pub fn anchorize(&mut self, s: &str) -> String {
```